Lib/cgi.py

```diff
@@ -638,6 +642,8 @@ def read_multi(self, environ, keep_blank_values, strict_parsing):
                          self.encoding, self.errors)
```

Comment:
ISTM that we should be propagating through the `max_num_fields` limit for sub-parts (and maybe reducing by the aggregate number of fields read so far).

Reply:
Good thought, I see what you mean there. Thinking about how to implement that.

Reply:
Okay, added that in 21e5e488a4 along with unit tests.